python/lsst/meas/algorithms/dynamicDetection.py

```diff
@@ -120,16 +120,20 @@ def calculateThreshold(self, exposure, seed, sigma=None):
 
         # Calculate new threshold
         fluxes = catalog["base_PsfFlux_flux"]
-        good = ~catalog["base_PsfFlux_flag"] & np.isfinite(fluxes)
+        area = catalog["base_PsfFlux_area"]
+        bg = catalog["base_LocalBackground_flux"]
+
+        good = (~catalog["base_PsfFlux_flag"] & ~catalog["base_LocalBackground_flag"] &
+                np.isfinite(fluxes) & np.isfinite(area) & np.isfinite(bg))
 
         if good.sum() < self.config.minNumSources:
             self.log.warn("Insufficient good flux measurements (%d < %d) for dynamic threshold calculation",
                           good.sum(), self.config.minNumSources)
             return Struct(multiplicative=1.0, additive=0.0)
 
-        bgMedian = np.median((fluxes/catalog["base_PsfFlux_area"])[good])
+        bgMedian = np.median((fluxes/area)[good])
 
-        lq, uq = np.percentile(fluxes[good], [25.0, 75.0])
+        lq, uq = np.percentile((fluxes - bg*area)[good], [25.0, 75.0])
```

Comment:
How does this change the multiplicative term?

Reply:
Oh I get it, you don't want variations in background to increase the estimate of the std-dev?